lib/AST/Decl.cpp

```diff
@@ -2311,7 +2320,12 @@ CanType ValueDecl::getOverloadSignatureType() const {
     if (isa<VarDecl>(this)) {
       defaultSignatureType = TupleType::getEmpty(getASTContext());
     } else {
-      defaultSignatureType = getInterfaceType()->getCanonicalType();
+      defaultSignatureType = mapSignatureFunctionType(
```

Comment:
Does it matter whether we map the type before or after we add the curried `self` param? It means that we won't strip `inout` from the `self` param but it wasn't clear to me whether `addCurriedSelfType` already deals with that.

Reply:
addCurriedSelfType() does not mark the self parameter as inout even if the storage is mutating. So I guess it doesn't matter if you add it before or after.